controllers/ocicluster_controller.go

```diff
@@ -112,35 +113,49 @@ func (r *OCIClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		logger.Error(err, "Couldn't get the clients for region")
 	}
 
+	helper, err := patch.NewHelper(ociCluster, r.Client)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to init patch helper")
+	}
+
+	clusterAccessor := scope.OCISelfManagedCluster{
+		OCICluster: ociCluster,
+	}
 	clusterScope, err = scope.NewClusterScope(scope.ClusterScopeParams{
 		Client:             r.Client,
 		Logger:             &logger,
 		Cluster:            cluster,
-		OCICluster:         ociCluster,
+		OCIClusterAccessor: clusterAccessor,
 		ClientProvider:     r.ClientProvider,
 		VCNClient:          clients.VCNClient,
 		LoadBalancerClient: clients.LoadBalancerClient,
 		IdentityClient:     clients.IdentityClient,
-		Region:             regionOverride,
+		RegionIdentifier:   regionOverride,
 	})
+	if err != nil {
+		logger.Error(err, "Couldn't create cluster scope")
+		return ctrl.Result{}, err
+	}
 
 	// Always close the scope when exiting this function so we can persist any OCICluster changes.
 	defer func() {
 		logger.Info("Closing cluster scope")
-		if err := clusterScope.Close(ctx); err != nil && reterr == nil {
+		conditions.SetSummary(ociCluster)
+
+		if err := helper.Patch(ctx, ociCluster); err != nil && reterr == nil {
```

Comment:
yes, I copy pasted the code from cluster.go so that we can re-use cluster.go for both managed and self managed clusters. You should hopefully see the same lines being deleted from there :)